CRM/Report/Form.php

```diff
@@ -4668,48 +4676,41 @@ protected function addLinkToRow(&$row, $baseUrl, $linkText, $value, $fieldName,
    * Generate temporary table to hold all contributions with permissioned FTs.
    *
    * @param object $query
-   * @param string $alias
    * @param bool $return
    *
    * @return string
+   *
+   * TODO: This needs to be removed in favour of new hook approach.
    */
-  public function getPermissionedFTQuery(&$query, $alias = NULL, $return = FALSE) {
+  public function getPermissionedFTQuery(&$query, $return = FALSE) {
     if (!CRM_Financial_BAO_FinancialType::isACLFinancialTypeStatus()) {
       return FALSE;
     }
-    $financialTypes = NULL;
-    CRM_Financial_BAO_FinancialType::getAvailableFinancialTypes($financialTypes);
-    if (empty($financialTypes)) {
+    CRM_Financial_BAO_FinancialType::getAvailableFinancialTypes($this->financialTypes);
+    if (strpos($query->_from, 'civicrm_contribution') === FALSE) {
+      return FALSE;
+    }
+    if (empty($this->financialTypes)) {
       $contFTs = "0";
       $liFTs = implode(',', array_keys(CRM_Contribute_Pseudoconstant::financialType()));
     }
     else {
-      $contFTs = $liFTs = implode(',', array_keys($financialTypes));
-    }
-    $temp = CRM_Utils_Array::value('civicrm_line_item', $query->_aliases);
-    if ($alias) {
-      $query->_aliases['civicrm_line_item'] = $alias;
-    }
-    elseif (!$temp) {
-      $query->_aliases['civicrm_line_item'] = 'civicrm_line_item_civireport';
+      $contFTs = $liFTs = implode(',', array_keys($this->financialTypes));
```

Comment:
While this is closed now, just a note that these variable names don't conform to coding standards.